tests/aggregation_tests.py

```diff
@@ -11,278 +11,282 @@
             [
-                (2, 'DE', '2016-09-26', 1L, 0.0, 0),
-                (2, 'US', '2016-09-26', 2L, 0.0, 0)   # 2 entries grouped by country
+                (datetime.datetime(2016, 9, 29, 0, 0), 2, 'DE', 3333L, 1, 0.0),
+                (datetime.datetime(2016, 9, 29, 0, 0), 2, 'US', 12252L, 2, 0.0)  
             ])
 
-    def test_group_by_risk(self):
-        # GIVEN 3 entries, of the same week, two of which have same risk type, but different countries
-        ntp_scan_csv = dedent('''\
+
+    def test_group_by_asn(self):
+        # GIVEN 3 entries of the same risk, country and day, but with different asn
+        scan_csv = dedent('''\
         ts,ip,risk_id,asn,cc
-        2016-09-28T00:00:01+00:00,190.81.134.82,7,4444,US
+        2016-09-29T00:00:01+00:00,190.81.134.82,2,12252,US
         2016-09-29T00:00:01+00:00,190.81.134.11,2,12252,US
-        2016-09-29T00:00:01+00:00,190.81.134.11,2,3333,DE
+        2016-09-29T00:00:01+00:00,190.81.134.34,2,3333,US
         ''')
-        self.cursor.copy_expert("COPY logentry from STDIN csv header", StringIO(ntp_scan_csv))
+        self.cursor.copy_expert("COPY logentry from STDIN csv header", StringIO(scan_csv))
 
-        # WHEN grouped entries get created
-        main.create_count()
-        main.create_count_by_country()
-        main.create_count_by_risk()
+        main.aggregate()
 
-        # THEN count_by_risk table should have 2 entries which get grouped, and one entry which stands alone
-        self.cursor.execute('select * from count_by_risk;')
+        # count table should have 2 entries - AS 12252 with count of 2, and AS 3333 with count of 1
+        self.cursor.execute('select * from count;')
         self.assertEqual(
             self.cursor.fetchall(),
             [
-                (7, '2016-09-26', 1L, 1L),
-                (2, '2016-09-26', 2L, 1L)   # 2 enreies grouped by risk
+                (datetime.datetime(2016, 9, 29, 0, 0), 2, 'US', 3333L, 1, 0.0),
+                (datetime.datetime(2016, 9, 29, 0, 0), 2, 'US', 12252L, 2, 0.0)
             ])
 
-class ScoresTestCase(unittest.TestCase):
-    # Test scores calculation.
-
-    def setUp(self):
-        # Patch main connection with the test one.
-        patch('main.connection', connection).start()
-
-        # Set isolation level to run CREATE DATABASE statement outside of transactions.
-        main.connection.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
-
-        self.cursor = main.connection.cursor()
 
-        # Recreate logentry table
-        try:
-            self.cursor.execute("DROP TABLE %s" % (main.tablename))
-        except:
-            pass
-        main.create_table()
-
-    def test_scores5(self):
-        # GIVEN 5 entries from different hosts, of the same risk and week, 2 from US, and 3 from DE
-        ntp_scan_csv = dedent('''\
+    def test_end_to_end(self):
+        # GIVEN 17 entries of the 2 different risk, day, country and asn
```

Comment:
I see that there are different IPs too. It would be useful to mention or mark entries that should be grouped